src/utils/charts/getChartData.js

```diff
@@ -36,32 +75,85 @@ getData.section8 = function (xRange, client, multiplier) {
   return data;
 };  // End getData.section8()
 
+/**
+ * @param {array} xRange All income values to be included.
+ */
+const insertBenefitData = function (xRange, clone, multiplier, benefitDatasets) {
+
+  // Don't loop if there's nothing to do
+  if (benefitDatasets.length === 0) {
+    return benefitDatasets;
+  }
+
+  // Otherwise, loop over incomes
+  for (var incomei = 0; incomei < xRange.length; incomei++) {
+    var income = xRange[ incomei ];
+    clone.future.earned = income / multiplier;
+
+    // Datasets need to be in correct order for benefits to
+    // affect each other corrrectly.
+    for (let benefiti = 0; benefiti < benefitDatasets.length; benefiti++) {
+
+      let dataset = benefitDatasets[ benefiti ],
+          name    = dataset.benefitName,
+          funcs   = getBenefit[ name ];
```

Comment:
@wacii wrote this, and it's about to be oudated by an unrelated change:
This inner loop and `getBenefit` are too strongly coupled. I would rather see `getBenefit[name]` resolve to a function that does both the subsidy calculation and the update of the client (when necessary).

Reply:
I like this idea, and it's one I'll work on, but maybe on a different PR? I'd like to get this fix through, even if it's just a temporary one, so that we have a tool that's working.